scripts/update_jobs.py

```
@@ -1376,8 +1376,23 @@ def deduplicate_jobs(jobs: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
 
 def has_new_grad_signal(title: str, signals: List[str]) -> bool:
     """Check if job title contains new grad signals"""
-    title_lower = title.lower()
-    return any(signal.lower() in title_lower for signal in signals)
+    #1. fast exit to save compute if not signals
+    if not signals:
+        return False
+    #2. Type guard; if not a string, not a signal.
+    elif not isinstance(title, str):
```

Ruff (0.15.5)

[warning] Unnecessary `elif` after `return` statement

Remove unnecessary `elif`

(RET505)